src/client/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -207,44 +293,39 @@ protected void InitializeHttpClient(HttpClientHandler httpClientHandler, params
             FirstMessageHandler = currentHandler;
             HttpClient = newClient;
             Type type = this.GetType();
-            HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(type.FullName,
-                GetClientVersion()));
+            HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(type.FullName, ClientVersion));
         }
 
-        /// <summary>
-        /// Sets the product name to be used in the user agent header when making requests
-        /// </summary>
-        /// <param name="productName">Name of the product to be used in the user agent</param>
-        public bool SetUserAgent(string productName)
+        public bool SetUserAgent()
         {
             if (!_disposed && HttpClient != null)
             {
                 // Clear the old user agent
                 HttpClient.DefaultRequestHeaders.UserAgent.Clear();
-                HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(productName, GetClientVersion()));
+                HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(FXVERSION, FrameworkVersion));
 
                 // Returns true if the user agent was set 
                 return true;
             }
 
             // Returns false if the HttpClient was disposed before invoking the method
             return false;
+            
         }
 
         /// <summary>
-        /// Sets the product name and version to be used in the user agent header when making requests
+        /// Sets the product name to be used in the user agent header when making requests
         /// </summary>
         /// <param name="productName">Name of the product to be used in the user agent</param>
-        /// <param name="version">Version of the product to be used in the user agent</param>
-        public bool SetUserAgent(string productName, string version)
+        public bool SetUserAgent(string productName)
         {
             if (!_disposed && HttpClient != null)
             {
                 // Clear the old user agent
```

Comment:
This comment seems to be wrong. Should it be clear the old user agent and set the FxVersion or is it needed at all?